zerver/views/webhooks/wordpress.py

```diff
@@ -0,0 +1,52 @@
+# Webhooks for external integrations.
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from django.http import HttpRequest, HttpResponse
+from zerver.models import Client, get_client, UserProfile
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+from six import text_type
+
+@api_key_only_webhook_view("Wordpress")
+@has_request_variables
+def api_wordpress_webhook(request, user_profile, client,
+                          stream=REQ(default="wordpress"),
+                          topic=REQ(default="WordPress Notification"),
+                          hook=REQ(default="WordPress Action"),
+                          post_title=REQ(default="New WordPress Post"),
+                          post_type=REQ(default="post"),
+                          post_url=REQ(default="WordPress Post URL"),
+                          display_name=REQ(default="New User Name"),
+                          user_email=REQ(default="New User Email"),
+                          user_login=REQ(default="Logged in User")):
+    # type: (HttpRequest, UserProfile, Client, text_type, text_type, text_type, text_type, text_type, text_type, text_type, text_type, text_type) -> HttpResponse
+
+    # remove trailing whitespace (issue for some test fixtures)
+    hook = hook.rstrip()
+    post_title = post_title.rstrip()
+    post_type = post_type.rstrip()
+    post_url = post_url.rstrip()
+    display_name = display_name.rstrip()
+    user_email = user_email.rstrip()
+    user_login = user_login.rstrip()
+
+    if hook == 'publish_post' or hook == 'publish_page':
+        data_template = 'New {type} published.\n[{title}]({url})'
```

Comment:
Modifying anything other than the static text (to change which fields are displayed, for example) would require also changing the line that uses the template. 

I think this could go either way. Someone modifying the static text of messages would be helped by having the templates be static values in one convenient location. Someone changing the fields used or adding new supported actions would be helped by having all the relevant code together. 

Are people doing their own localizations for provided integrations? I have no idea. I hope that the set of provided actions is sufficient to cover most cases, and there is less need to add new ones. 

Based on localization, I'll add your suggested change. I'm not aware of a plan to internationalize webhooks, so without that it's probably the more likely reason someone would wish to modify this code in the future.